include/sdf/Light.hh

```diff
@@ -268,6 +268,11 @@ namespace sdf
     private: void SetPoseRelativeToGraph(
         sdf::ScopedGraph<PoseRelativeToGraph> _graph);
 
+    /// \brief Fill the provided _elem with data from this light.
+    /// \param[out] _elem SDF element point to populate
+    /// \return True if successful.
+    public: bool PopulateElement(sdf::ElementPtr _elem) const;
```

Comment:
Can you add documentation that the user first needs to:
  * Allocate memory for `_elem`. i.e, just creating an `sdf::ElementPtr` object and calling this function with it would crash the program.
  * Initialize `_elem` with `sdf::initFile` using the correct description file. In this case, `light.sdf`.

To avoid confusion, would it be better for the function to do both and return an `sdf::ElementPtr` instead?

Reply:
> To avoid confusion, would it be better for the function to do both and return an sdf::ElementPtr instead?

do you mean something like `sdf::ElementPtr Element() const` that creates, initializes, and returns the sdf::ElementPtr?

Reply:
Yes, but named differently so it doesn't collide with the existing `Element()` functions in DOM objects. Maybe `BuildElement` or `CreateElement`?

Reply:
ok sounds good. How about `ToElement`? same idea as the other `ToString()` functions in sdformat?

Reply:
Yeah, `ToElement` sounds good!

Reply:
done. 5b68f18

Reply:
I'll create a separate PR to update the other PopulateElement APIs introduced in https://github.com/ignitionrobotics/sdformat/pull/753